codex/utils/exceptions.nim

```diff
@@ -4,4 +4,18 @@ proc msgDetail*(e: ref CatchableError): string =
   var msg = e.msg
   if e.parent != nil:
     msg = fmt"{msg} Inner exception: {e.parent.msg}"
-  return msg
\ No newline at end of file
+  return msg
+
+template launderBare*(body: untyped): untyped =
```

Comment:
see `handleException`: https://status-im.github.io/nim-chronos/error_handling.html#the-exception-type

Reply:
Nice!

Reply:
Yeah, this has been a discussion point I had raised earlier, but my usage of this is quite specific. If you look closely, the only place where actually use this is [in `parseJson`](https://github.com/codex-storage/nim-codex/pull/673/files#diff-5527d689b628357c67c68656973313c0ceb316441b38fd493b187e909d017b68R225), the reason being that the  [1.6.x stream API is annotated to raise Exception](https://github.com/nim-lang/Nim/blob/version-1-6/lib/pure/streams.nim#L113), and through a somewhat long-winded path, [`parseJson` calls `close(Stream)`](https://github.com/nim-lang/Nim/blob/version-1-6/lib/pure/json.nim#L954). This [is no longer true in 2.0.0](https://github.com/nim-lang/Nim/blob/version-2-0/lib/pure/streams.nim#L117), so the issue is temporary.

Since we have all calls to `parseJson` neatly encapsulated under `fromJson`, I decided to simply "backport" the behavior of `2.x` into our code and thus avoid contaminating all callers with `handleException: true` when `parseJson` should not be raising `Exception` in the first place.

Reply:
fair.